data/scripts.js

```diff
@@ -1170,6 +1170,9 @@ let BattleScripts = {
 		// {gigantamax?: string, maxMoves: {[k: string]: string} | null}[]
 		if (!skipChecks) {
 			if (!pokemon.canDynamax) return;
+			if (pokemon.template.isMega || pokemon.template.isPrimal || pokemon.template.forme === "Ultra") {
```

Comment:
I ended up getting rid of the check for ditto when it has transformed into Zacian, Zamazenta, or Eternatus. because when I did further testing, ditto was not able to Dynamax if it had transformed into those three.